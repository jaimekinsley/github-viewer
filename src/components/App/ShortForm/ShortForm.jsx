import React from 'react';
import PropTypes from 'prop-types';

const ShortForm = ({ username, onChange, onSubmit }) => (
  <form onSubmit={onSubmit}>
    <input type="text" value={username} onChange={onChange} />
    <button>Submit</button>
  </form>
);

ShortForm.propTypes = {
  username: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default ShortForm;
