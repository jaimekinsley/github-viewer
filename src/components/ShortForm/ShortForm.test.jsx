import React from 'react';
import { shallow } from 'enzyme';
import ShortForm from './ShortForm';

describe('ShortForm component', () => {
  it('renders ShortForm', () => {
    const wrapper = shallow(<ShortForm username="jaimekinsley" onChange={() =>{}} onSubmit={() => {}}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
