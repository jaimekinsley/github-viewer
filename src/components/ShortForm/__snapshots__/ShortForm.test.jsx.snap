// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShortForm component renders ShortForm 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value="jaimekinsley"
  />
  <button>
    Submit
  </button>
</form>
`;
